app/services/bulkrax/factory_class_finder.rb

```diff
@@ -66,27 +66,68 @@ def find
       entry.default_work_type.constantize
     end
 
+    private
+
     ##
     # @api private
     # @return [String]
     def name
-      fc = if entry.parsed_metadata&.[]('model').present?
-             Array.wrap(entry.parsed_metadata['model']).first
-           elsif entry.importerexporter&.mapping&.[]('work_type').present?
-             # Because of delegation's nil guard, we're reaching rather far into the implementation
-             # details.
-             Array.wrap(entry.parsed_metadata['work_type']).first
-           else
-             entry.default_work_type
-           end
-
-      # Let's coerce this into the right shape; we're not mutating the string because it might well
-      # be frozen.
-      fc = fc.tr(' ', '_')
-      fc = fc.downcase if fc.match?(/[-_]/)
-      fc.camelcase
+      # Try each strategy in order until one returns a value
+      fc = find_factory_class_name || entry.default_work_type
+
+      # Normalize the string format
+      normalize_class_name(fc)
     rescue
       entry.default_work_type
     end
+
+    ##
+    # Try each strategy in sequence to find a factory class name
+    # @return [String, nil] the factory class name or nil if none found
+    def find_factory_class_name
```

Comment:
My colleague @sandbergja came up with a pattern I liked for something similar - basically make each strategy its own class, then have the classes have a consistent api of `.eligible?`, then you could select all eligible factories. 

See [orangelight requests router](https://github.com/pulibrary/orangelight/blob/main/app/models/requests/router.rb) - just something to think about for the future!